src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -216,18 +288,58 @@ private static bool CommandNamesEqual(ToolCommandName[] left, ToolCommandName[]
             return left.SequenceEqual(right);
         }
 
-        [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
         private class SerializableLocalToolsManifest
         {
-            [DefaultValue(1)]
-            [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
-            public int Version { get; set; }
+            public int? Version { get; set; }
 
             public bool? IsRoot { get; set; }
 
-            [JsonProperty(Required = Required.Always)]
-            // The dictionary's key is the package id
+            /// <summary>
+            /// The dictionary's key is the package id
+            /// this field could be null
+            /// </summary>
             public Dictionary<string, SerializableLocalToolSinglePackage> Tools { get; set; }
+
+            public string ToJson()
+            {
+                var state = new JsonWriterState(options: new JsonWriterOptions { Indented = true });
+                var arrayBufferWriter = new ArrayBufferWriter<byte>();
+                var writer = new Utf8JsonWriter(arrayBufferWriter, state);
+
+                writer.WriteStartObject();
+
+                if (Version.HasValue)
+                {
+                    writer.WriteNumber(propertyName: JsonPropertyVersion, value: Version.Value);
+                }
+
+                if (IsRoot.HasValue)
+                {
+                    writer.WriteBoolean(JsonPropertyIsRoot, IsRoot.Value);
+                }
+
+                writer.WriteStartObject(JsonPropertyTools);
+
+                foreach (var tool in Tools)
```

Comment:
How many Tools can there be? Essentially how big is this JSON file that is being written going to be (in the worst case)? Will it fit within 1 MB? `Int32.MaxValue`?

Reply:
it should not be a big number. Tools required to work on a project. I don't think people can handle more than 20 for a project.

But is there a cap? what happens if it go over 1MB. Is there a nob to change that? Even though it is not for this scenario, but i am surprised. A lot of json can easily go over 1MB

Reply:
Its fine for you to go over 1 MB, the ArrayBufferWriter just ends up allocating at that point (you could flush/clear to avoid that, which is why I asked). If you are writing over 2 GB, you likely want to flush to your underlying output stream and clear your ArrayBufferWriter to avoid overflowing.